sensor/FamilyProtectPlugin.js

```diff
@@ -0,0 +1,193 @@
+/*    Copyright 2016 Firewalla LLC
+ *
+ *    This program is free software: you can redistribute it and/or  modify
+ *    it under the terms of the GNU Affero General Public License, version 3,
+ *    as published by the Free Software Foundation.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+'use strict';
+
+const log = require('../net2/logger.js')(__filename);
+
+const Sensor = require('./Sensor.js').Sensor;
+
+const sem = require('../sensor/SensorEventManager.js').getInstance();
+
+const extensionManager = require('./ExtensionManager.js')
+
+const f = require('../net2/Firewalla.js');
+
+const userConfigFolder = f.getUserConfigFolder();
+const dnsmasqConfigFolder = `${userConfigFolder}/dns`;
+const updateInterval = 3600 * 1000 // once per hour
+
+const fs = require('fs');
+const Promise = require('bluebird');
+Promise.promisifyAll(fs);
+
+const DNSMASQ = require('../extension/dnsmasq/dnsmasq.js');
+const dnsmasq = new DNSMASQ();
+
+const exec = require('child-process-promise').exec;
+
+const fc = require('../net2/config.js');
+
+class FamilyProtectPlugin extends Sensor {
+    async run() {
+        this.systemSwitch = false;
+        this.adminSystemSwitch = false;
+        this.enabledMacAddresses = {};
+        extensionManager.registerExtension("family", this, {
+            applyPolicy: this.applyPolicy,
+            start: this.start,
+            stop: this.stop
+        });
+        await exec(`mkdir -p ${dnsmasqConfigFolder}`);
+        if (fc.isFeatureOn("family_protect")) {
+            await this.globalOn();
+        } else {
+            await this.globalOff();
+        }
+        fc.onFeature("family_protect", async (feature, status) => {
```

Comment:
Also need to check fc.isFeatureOn("family_protect") during startup to enable/disable feature immediately.